tools/internal_ci/linux/grpc_xds_k8s_lb_python.sh

```diff
@@ -133,15 +132,30 @@ run_test() {
 main() {
   local script_dir
   script_dir="$(dirname "$0")"
-  # shellcheck source=tools/internal_ci/linux/grpc_xds_k8s_install_test_driver.sh
-  source "${script_dir}/grpc_xds_k8s_install_test_driver.sh"
+  
+  # Clone the test driver from the master branch using an external script.
+  # shellcheck source=tools/internal_ci/linux/grpc_xds_k8s_clone_driver_repo.sh
+  source "${script_dir}/grpc_xds_k8s_clone_driver_repo.sh"
+  clone_test_driver
```

Comment:
That was my first thought as well, but Sergii had done some initial work with the clone approach so I went with that. Performance is not really a concern here and it doesn't hurt to have the whole master branch available.

Reply:
I think by going with a simple curl, we can save managing an extra file in java/go/node repos. This will be a significant improvement for reducing coupling between repos. E.g., in the worst scenario, if there is a small improvement we need to do in the clone logic, we will need to patch all release branches across repos. I agree performance is not the main issue here, my suggestion is more about reducing future maintenance toil.

@sergiitk What do you think about using `curl` to download the xds_k8s utility functions?

Reply:
Yes, my initial work was from some months ago, and I don't fully remember the reasoning for git clone vs curl. I remember talking to @ejona86 about it, and we concluded on cloning. Now, as it introduces an extra file to manage, I like the idea the curl oneliner.

Reply:
I initially liked the clone approach as it would allow us to potentially use other files in the future. But I can't really think what files those would be, so I'll switch to curl and eliminate the new clone script.

Reply:
I agree that Git clone is still necessary, but it will happen after the downloaded entrypoint script is sourced. It would look like following:

```diff
  # shellcheck source=buildscripts/kokoro/xds-k8s-install-test-driver.sh
- source "${script_dir}/xds-k8s-install-test-driver.sh"
+ source /dev/stdin <<< "$(curl https://raw.githubusercontent.com/grpc/grpc/master/tools/internal_ci/linux/grpc_xds_k8s_install_test_driver.sh)"

  set -x
  if [[ -n "${KOKORO_ARTIFACTS_DIR}" ]]; then
    kokoro_setup_test_driver "${GITHUB_REPOSITORY_NAME}"
  else
    local_setup_test_driver "${script_dir}"
  fi
```

Reply:
The install script is now sourced and I removed the clone script, take another look please.

@sergiitk
@lidizheng